Source/Scene/CreditDisplay.js

```diff
@@ -77,113 +76,275 @@ define([
             credit = displayedImageCredits[i];
             if (defined(credit)) {
                 removeCreditDomElement(credit);
             }
         }
+        var displayedLightboxCredits = creditDisplay._displayedCredits.lightboxCredits;
+        for (i = 0; i < displayedLightboxCredits.length; i++) {
+            credit = displayedLightboxCredits[i];
+            if (defined(credit)) {
+                removeCreditDomElement(credit);
+            }
+        }
+    }
+
+    function styleLightboxContainer(that) {
+        var lightboxCredits = that._lightboxCredits;
+        var width = that.viewport.clientWidth;
+        var height = that.viewport.clientHeight;
+        if (width !== that._lastViewportWidth) {
+            if (width < mobileWidth) {
+                lightboxCredits.style.border = 'none';
+                lightboxCredits.style.borderRadius = '0';
+                lightboxCredits.style.maxWidth = 'initial';
+                lightboxCredits.style.marginTop = '0';
+                lightboxCredits.style.height = '100%';
+                lightboxCredits.style.width = '100%';
+            } else {
+                lightboxCredits.style.border = '1px solid #444';
+                lightboxCredits.style.borderRadius = '5px';
+                lightboxCredits.style.maxWidth = '370px';
+                lightboxCredits.style.height = 'initial';
+                lightboxCredits.style.width = 'initial';
+                lightboxCredits.style.marginTop = Math.floor((height - lightboxCredits.clientHeight) * 0.5) + 'px';
+            }
+            that._lastViewportWidth = width;
+        }
+
+        if (width >= mobileWidth && height !== that._lastViewportHeight) {
+            lightboxCredits.style.marginTop = Math.floor((height - lightboxCredits.clientHeight) * 0.5) + 'px';
+            that._lastViewportHeight = height;
+        }
     }
 
     /**
      * The credit display is responsible for displaying credits on screen.
      *
      * @param {HTMLElement} container The HTML element where credits will be displayed
      * @param {String} [delimiter= ' • '] The string to separate text credits
+     * @param {HTMLElement} [viewport=document.body] The HTML element that will contain the credits popup
      *
      * @alias CreditDisplay
      * @constructor
      *
      * @example
      * var creditDisplay = new Cesium.CreditDisplay(creditContainer);
      */
-    function CreditDisplay(container, delimiter) {
+    function CreditDisplay(container, delimiter, viewport) {
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(container)) {
-            throw new DeveloperError('credit container is required');
-        }
+        Check.defined('container', container);
         //>>includeEnd('debug');
+        var that = this;
+
+        viewport = defaultValue(viewport, document.body);
+
+        var lightbox = document.createElement('div');
+        lightbox.className = 'cesium-credit-lightbox-overlay';
```

Comment:
Since we add a className here, do we need immediate styles below?  Or can the styles just be picked up from stylesheets?  I ask because it's easier for an app's stylesheets to override Cesium stylesheets than to override immediate styles.